rpc/net.go

```diff
@@ -73,6 +88,71 @@ func (s *NetService) Peers(_ *http.Request, _ *interface{}, resp *rpctypes.Peers
 	return nil
 }
 
+// Pairs returns all currently available pairs from offers of all peers
+func (s *NetService) Pairs(_ *http.Request, req *rpctypes.PairsRequest, resp *rpctypes.PairsResponse) error {
+	if s.isBootnode {
+		return errUnsupportedForBootnode
+	}
+
+	peerIDs, err := s.discover(req)
```

Comment:
this only discovers peers advertising the coin inside `req.Provides`, but the function comment says it returns "returns all currently available pairs", but this will only return pairs that include the `req.Provides` coin